packages/plugin-hardhat/test/beacon-constructor.js

```diff
@@ -10,19 +10,19 @@ test('new beacon - do not redeploy with same args', async t => {
   const { WithConstructor } = t.context;
 
   const beacon1 = await upgrades.deployBeacon(WithConstructor, { constructorArgs: [17] });
-  const implementation1 = await upgrades.beacon.getImplementationAddress(beacon1.address);
+  const implementation1 = await upgrades.beacon.getImplementationAddress(await beacon1.getAddress());
   const proxy1 = await upgrades.deployBeaconProxy(beacon1, WithConstructor);
-  t.is((await proxy1.value()).toNumber(), 17);
+  t.is(Number(await proxy1.value()), 17);
```

Comment:
Previously we used `toNumber` because the `BigNumber` objects were a pain to work with, in particular not comparable using `t.is`. One of the biggest selling points of Ethers v6 is the use of native BigInt. This means we can now write:

```suggestion
  t.is(await proxy1.value(), 17n);
```

where `17n` is a bigint literal.

We should be using this new ability because it really improves test readability.